File: components/physics/src/test/java/org/storm/physics/math/PointTest.java
package org.storm.physics.math;

import org.junit.jupiter.api.Test;
import org.storm.physics.math.geometry.Point;

import static org.junit.jupiter.api.Assertions.assertEquals;


public class PointTest {

  @Test
  public void testTranslate() {
    Point point = new Point(1, 2);
    assertEquals(1, point.getX(), 0);
    assertEquals(2, point.getY(), 0);

    point.translate(-2, 2);
    assertEquals(-1, point.getX(), 0);
    assertEquals(4, point.getY(), 0);
  }

  @Test
  public void testDistanceTo() {
    Point p = new Point(1, 1);
    Point q =  new Point(5, 4);

    assertEquals(5.0, p.getDistance(q), 0.0);
    assertEquals(5.0, q.getDistance(p), 0.0);
  }

}

File: components/physics/src/test/kotlin/org/storm/physics/math/PointTest.kt
package org.storm.physics.math

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.storm.core.render.geometry.Point
import org.storm.physics.math.extensions.getDistance

class PointTest {

    @Test
    fun testTranslate() {
        val point = Point(1.0, 2.0)
        assertEquals(1.0, point.x, 0.0)
        assertEquals(2.0, point.y, 0.0)

        point.translate(-2.0, 2.0)
        assertEquals(-1.0, point.x, 0.0)
        assertEquals(4.0, point.y, 0.0)
    }

    @Test
    fun testDistanceTo() {
        val p = Point(1.0, 1.0)
        val q = Point(5.0, 4.0)

        assertEquals(5.0, p.getDistance(q), 0.0)
        assertEquals(5.0, q.getDistance(p), 0.0)
    }

}
